packages/x-date-pickers/src/TimePicker/TimePicker.tsx

```diff
@@ -26,7 +26,7 @@ const TimePicker = React.forwardRef(function TimePicker<TDate>(
     return <DesktopTimePicker ref={ref} {...other} />;
   }
 
-  return <MobileTimePicker ref={ref} {...other} />;
+  return <MobileTimePicker ref={ref} {...(other as MobileTimePickerProps<TDate>)} />;
```

Comment:
If you have better ideas of how to avoid TS complaining of receiving props with view generic as `TimeViewWithMeridiem` instead of expected `TimeView` I am open to suggestions. 🙈 🤷